File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n=len(cardPoints)
        if(n==k):
            return sum(cardPoints)
        l_sum,maxsum=0,0
        r_sum=0
        for i in range(k):
            l_sum+=cardPoints[i]
            maxsum=max(maxsum,l_sum)
        r_len=n-1
        for i in range(k-1,-1,-1):
            l_sum-=cardPoints[i]
            r_sum+=cardPoints[r_len]
            r_len-=1
            maxsum=max(maxsum,(l_sum+r_sum))  
        return maxsum      

        

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] arr, int k) {
       int lsum=0,rsum=0, maxlen=0;
        for(int i=0;i<k;i++){
            lsum+=arr[i];
            maxlen=Math.max(maxlen,lsum);
        }
        int rightidx=arr.length-1;
        for(int i=k-1;i>=0;i--){
            lsum-=arr[i];
            rsum+=arr[rightidx];
            rightidx--;
            maxlen=Math.max(maxlen,(rsum+lsum));
        }
        return maxlen; 
    }
}